mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -719,8 +723,15 @@ static LogicalResult printOperation(CppEmitter &emitter, emitc::CastOp castOp) {
   if (failed(emitter.emitAssignPrefix(op)))
     return failure();
   os << "(";
-  if (failed(emitter.emitType(op.getLoc(), op.getResult(0).getType())))
-    return failure();
+  // Cast of lvalues return lvalues and therefore references.
```

Comment:
The comment doesn't seem to match the `if`: We check for an attribute, not for being an lvalue

Reply:
Right, this is a remnant from the previous attempt that introduced `returnsLValue`. I wasn't happy with the incursion of semantics into the printer... thanks for pointing this out!